java/runtime/src/main/java/io/ray/runtime/task/TaskExecutor.java

```diff
@@ -141,7 +140,7 @@ private void throwIfDependencyFailed(Object arg) {
         } else {
           result = rayFunction.getConstructor().newInstance(args);
         }
-      } catch (InvocationTargetException e) {
+      } catch (Exception e) {
```

Comment:
```suggestion
      } catch (Throwable e) {
```

Reply:
And please change
```
if (e.getCause() != null) {
    throw e.getCause();
}
```
to
```
if (e instanceof InvocationTargetException && e.getCause() != null) {
    throw e.getCause();
}
```

Reply:
`Throwable` seems too broad to me. It should be handled at outer block.

Reply:
Normally we should not catch such broad exception such as `IOExcepiton` or `Exception`. Is it necessary? or could we narrow it down?

Reply:
@WangTaoTheTonic Yes you're right by principle, but here we just log the function and arguments then re-throw it.
And another reason is if we don't catch `Exception`, we have to catch these 4 specific exceptions: InstantiationException, IllegalAccessException, IllegalArgumentException, reflect.InvocationTargetException.

Reply:
How about we trying to catching these 4 kinds of exceptions if there's only 4?

Reply:
I removed this line.